src/backend/BaseRecordComponent.cpp

```diff
@@ -64,10 +64,17 @@ BaseRecordComponent::availableChunks()
         Offset offset( m_dataset->extent.size(), 0 );
         return ChunkTable{ { std::move( offset ), m_dataset->extent } };
     }
+
+    // set dirty, so Series::flush will open the file if needed
+    this->dirty() = true;
+    this->seriesFlush();
+    this->dirty() = false;
+
```

Comment:
I'm a bit doubtful about this..
1. It makes the `availableChunks` call collective, we should at least document that
2. `availableChunks` will now silently flush the whole Series. This will break user code that peruses our new flushing logic and its guarantees of no writes occurring until flushing. If we really want to do this, this function must not do anything more than merely opening those files.